From: Colin Walters <walters@verbum.org>
Date: Fri, 25 Jun 2010 09:37:06 -0400
Subject: [PATCH] Don't attempt to build dtrace support on OS X

Apparently the OS X "dtrace" command is different from the Sun one,
which is what Linux supports.  Since I don't have access to an OS X
machine to test build patches on, simply disable dtrace on OS X
for now.

https://bugzilla.gnome.org/show_bug.cgi?id=622697
--- configure.in.orig
+++ configure.in
@@ -2789,17 +2789,21 @@ AC_ARG_ENABLE([dtrace],
 have_dtrace=no
 AC_MSG_CHECKING([whether to include dtrace tracing support])
 if test "x$enable_dtrace" != xno; then
-  AC_MSG_RESULT([yes])
-  AC_CHECK_PROGS(DTRACE, dtrace)
-  if test -z "$DTRACE"; then
-    if test "x$enable_dtrace" = xyes; then
-      AC_MSG_ERROR([dtrace not found])
+  if test x$glib_have_carbon = xyes; then
+    AC_MSG_RESULT([no (not yet compatible with MacOS dtrace)])
+  else 
+    AC_MSG_RESULT([yes])
+    AC_CHECK_PROGS(DTRACE, dtrace)
+    if test -z "$DTRACE"; then
+      if test "x$enable_dtrace" = xyes; then
+        AC_MSG_ERROR([dtrace not found])
+      fi
     fi
+    AC_CHECK_HEADER([sys/sdt.h],have_dtrace=yes,
+                    [if test "x$enable_dtrace" = xyes; then
+                      AC_MSG_ERROR([dtrace support needs sys/sdt.h header])
+                     fi])
   fi
-  AC_CHECK_HEADER([sys/sdt.h],have_dtrace=yes,
-                  [if test "x$enable_dtrace" = xyes; then
-                     AC_MSG_ERROR([dtrace support needs sys/sdt.h header])
-                   fi])
 else
   AC_MSG_RESULT([no])
 fi
